packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -141,7 +142,42 @@ export default abstract class AddressBasedPaymentNetwork<
     };
   }
 
-  protected abstract isValidAddress(_address: string, _networkName?: string): boolean;
+  protected isValidAddress(address: string, networkName?: string): boolean {
+    if (networkName) {
+      return this.isValidAddressForNetwork(address, networkName);
+    }
+    return this.supportedNetworks.some((network) =>
+      this.isValidAddressForNetwork(address, network),
+    );
```

Comment:
maybe outside the scope of this PR but I feel the network should be mandatory

Reply:
Yes I agree with you. It's a major refacto as there is a lot of usage of this method without the second argument in our code, but it should be doable as we should know the network in advance in all those cases.